import Firebase is not sufficient to define the FirebaseApp symbol in my build environment.

<!-- DO NOT DELETE
validate_template=true
template_path=.github/ISSUE_TEMPLATE/bug_report.md
-->

### [REQUIRED] Step 1: Describe your environment
  * Xcode version: 11.6 (11E708)
  * Firebase SDK version: 6.31.0
  * Firebase Component: Analytics
  * Component version: 6.8.0
  * Installation method: CocoaPods
  * macOS version: 10.15.6 (19G2021)

### [REQUIRED] Step 2: Describe the problem

'import Firebase' is no longer sufficient to define the FirebaseApp symbol in my build environment.  Using Xcode 11.6, with Firebase 6.31.0, I have to import FirebaseCore to avoid getting: "error: use of unresolved identifier 'FirebaseApp'".  I was previously using Firebase 6.27.0 on this same build machine without issue when only importing Firebase.

I believe this is the same issue as #6066.  The comments in that issue indicate that 'import Firebase' should be sufficient and that importing the specific component (or firebase core) is not needed.  I have run through the proposed resolution in that issue (quit Xcode, remove pods dir, remove xcworkspace dir, rerun pod install, open newly created xcworkspace).  This does not resolve the issue in my project.

I am able to reproduce the same issue with the [QuickStart sample](https://github.com/firebase/quickstart-ios) for analytics.

This problem does not occur on my other build machine, which has the same firebase environment, but is running Xcode 11.5 (11E608c) on macOS 10.15.4 (19E266).

#### Steps to reproduce:

Reproducible with [QuickStart sample](https://github.com/firebase/quickstart-ios): 
$ git clone https://github.com/firebase/quickstart-ios.git
$ cd quickstart-ios/analytics/
$ pod install --repo-update
$ open AnalyticsExample.xcworkspace

Building with the Debug configuration for physical device running iOS 13 produces:

```
Prepare build
note: Using new build system
note: Building targets in parallel
note: Planning build
note: Constructing build description

Build target nanopb of project Pods with configuration Debug
Build target Pods-AnalyticsExample of project Pods with configuration Debug
Build target FirebaseCoreDiagnostics of project Pods with configuration Debug
Build target PromisesObjC of project Pods with configuration Debug
Build target GoogleDataTransport of project Pods with configuration Debug
Build target FirebaseCore of project Pods with configuration Debug
Build target AnalyticsExample of project AnalyticsExample with configuration Debug

CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler (in target 'AnalyticsExample' from project 'AnalyticsExample')
    cd /Users/user/workspaces/quickstart-ios/analytics
    export DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer
    export SDKROOT=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.6.sdk

<...build removed for brevity...>

/Users/user/workspaces/quickstart-ios/analytics/AnalyticsExample/AppDelegate.swift:23:5: error: use of unresolved identifier 'FirebaseApp'
    FirebaseApp.configure()
    ^~~~~~~~~~~

Build target FirebaseInstallations of project Pods with configuration Debug
Build target GoogleUtilities of project Pods with configuration Debug

Build failed    8/26/20, 11:45 PM    12.7 seconds
```

#### Relevant Code:

From [QuickStart sample](https://github.com/firebase/quickstart-ios)/analytics/AnalyticsExample/AppDelegate.swift:
```
import UIKit
import Firebase

@UIApplicationMain
class AppDelegate: UIResponder, UIApplicationDelegate {
  func application(_ application: UIApplication,
                   didFinishLaunchingWithOptions launchOptions: [UIApplication
                     .LaunchOptionsKey: Any]?) -> Bool {
    FirebaseApp.configure()
    return true
  }
```

--- a/CoreOnly/Sources/Firebase.h
+++ b/CoreOnly/Sources/Firebase.h
@@ -12,45 +12,30 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// The module qualified imports are for CocoaPods and the simple file names
-// for Swift Package Manager.
+#import <FirebaseCore/FirebaseCore.h>
 
 #if !defined(__has_include)
   #error "Firebase.h won't import anything if your compiler doesn't support __has_include. Please \
           import the headers individually."
 #else
-  #if __has_include(<FirebaseCore/FirebaseCore.h>)
-    #import <FirebaseCore/FirebaseCore.h>
-  #elif __has_include("FirebaseCore.h")
-    #import "FirebaseCore.h"
-  #endif
-
   #if __has_include(<FirebaseAnalytics/FirebaseAnalytics.h>)
     #import <FirebaseAnalytics/FirebaseAnalytics.h>
   #endif
 
-  #if __has_include(<FirebaseAuth/FirebaseAuth.h>)
-    #import <FirebaseAuth/FirebaseAuth.h>
-  #elif __has_include("FirebaseAuth.h")
-    #import "FirebaseAuth.h"
-  #endif
-
   #if __has_include(<FirebaseAppDistribution/FirebaseAppDistribution.h>)
     #import <FirebaseAppDistribution/FirebaseAppDistribution.h>
-  #elif  __has_include("FirebaseAppDistribution.h")
-    #import "FirebaseAppDistribution.h"
+  #endif
+
+  #if __has_include(<FirebaseAuth/FirebaseAuth.h>)
+    #import <FirebaseAuth/FirebaseAuth.h>
   #endif
 
   #if __has_include(<FirebaseCrashlytics/FirebaseCrashlytics.h>)
     #import <FirebaseCrashlytics/FirebaseCrashlytics.h>
-  #elif  __has_include("FirebaseCrashlytics.h")
-    #import "FirebaseCrashlytics.h"
   #endif
 
   #if __has_include(<FirebaseDatabase/FirebaseDatabase.h>)
     #import <FirebaseDatabase/FirebaseDatabase.h>
-  #elif __has_include("FirebaseDatabase.h")
-    #import "FirebaseDatabase.h"
   #endif
 
   #if __has_include(<FirebaseDynamicLinks/FirebaseDynamicLinks.h>)
@@ -62,20 +47,14 @@
 Firebase Dynamic Links works as intended."
       #endif // #ifndef FIREBASE_ANALYTICS_SUPPRESS_WARNING
     #endif
-  #elif __has_include("FirebaseDynamicLinks.h")
-    #import "FirebaseDynamicLinks.h"
   #endif
 
   #if __has_include(<FirebaseFirestore/FirebaseFirestore.h>)
     #import <FirebaseFirestore/FirebaseFirestore.h>
-  #elif __has_include("FirebaseFirestore.h")
-    #import "FirebaseFirestore.h"
   #endif
 
   #if __has_include(<FirebaseFunctions/FirebaseFunctions.h>)
     #import <FirebaseFunctions/FirebaseFunctions.h>
-  #elif __has_include("FirebaseFunctions.h")
-    #import "FirebaseFunctions.h"
   #endif
 
   #if __has_include(<FirebaseInAppMessaging/FirebaseInAppMessaging.h>)
@@ -87,8 +66,6 @@ Firebase Dynamic Links works as intended."
 Firebase In App Messaging works as intended."
       #endif // #ifndef FIREBASE_ANALYTICS_SUPPRESS_WARNING
     #endif
-  #elif __has_include("FirebaseInAppMessaging.h")
-    #import "FirebaseInAppMessaging.h"
   #endif
 
   #if __has_include(<FirebaseInstanceID/FirebaseInstanceID.h>)
@@ -104,8 +81,6 @@ Firebase In App Messaging works as intended."
 Firebase Messaging works as intended."
       #endif // #ifndef FIREBASE_ANALYTICS_SUPPRESS_WARNING
     #endif
-  #elif __has_include("FirebaseMessaging.h")
-    #import "FirebaseMessaging.h"
   #endif
 
   #if __has_include(<FirebaseMLCommon/FirebaseMLCommon.h>)
@@ -180,14 +155,10 @@ Firebase Performance works as intended."
 Firebase Remote Config works as intended."
       #endif // #ifndef FIREBASE_ANALYTICS_SUPPRESS_WARNING
     #endif
-  #elif __has_include("FirebaseRemoteConfig.h")
-    #import "FirebaseRemoteConfig.h"
   #endif
 
   #if __has_include(<FirebaseStorage/FirebaseStorage.h>)
     #import <FirebaseStorage/FirebaseStorage.h>
-  #elif __has_include("FirebaseStorage.h")
-    #import "FirebaseStorage.h"
   #endif
 
   #if __has_include(<GoogleMobileAds/GoogleMobileAds.h>)

--- a/FirebaseCore/CHANGELOG.md
+++ b/FirebaseCore/CHANGELOG.md
@@ -1,3 +1,7 @@
+# Unreleased
+- [fixed] Sporadic missing FirebaseApp symbol build issue introduced in Firebase 6.28.0. (#6341)
+
+
 # v6.10.1 -- M78
 - [added] Beta release of Swift Package Manager. Details
   [here](https://github.com/firebase/firebase-ios-sdk/blob/master/SwiftPackageManager.md). (#3136)

